tests/cursor/bug1529-001.phpt

```diff
@@ -0,0 +1,106 @@
+--SKIPIF--
+<?php if (!function_exists('pcntl_fork')) { die('skip pcntl_fork() not available'); } ?>
+<?php require __DIR__ . "/../utils/basic-skipif.inc"; ?>
+<?php skip_if_not_live(); ?>
+<?php skip_if_not_clean(); ?>
+--FILE--
+<?php
+require_once __DIR__ . "/../utils/basic.inc";
+
+class CommandLogger implements MongoDB\Driver\Monitoring\CommandSubscriber
+{
+    private $pid;
+
+    public function __construct()
+    {
+        $this->pid = getmypid();
+    }
+
+    public function commandStarted(MongoDB\Driver\Monitoring\CommandStartedEvent $event)
+    {
+        $command = $event->getCommand();
+        $commandName = $event->getCommandName();
+        $process = $this->pid === getmypid() ? 'Parent' : 'Child';
+
+        if ($commandName === 'find' || $commandName === 'getMore') {
+            printf("%s executes %s with batchSize: %d\n", $process, $commandName, $command->batchSize);
+            return;
+        }
+
+        printf("%s executes %s\n", $process, $commandName);
+    }
+
+    public function commandSucceeded(MongoDB\Driver\Monitoring\CommandSucceededEvent $event)
+    {
+    }
+
+    public function commandFailed(MongoDB\Driver\Monitoring\CommandFailedEvent $event)
+    {
+    }
+}
+
+$keyVaultClient = new MongoDB\Driver\Manager(URI, [], ['disableClientPersistence' => true]);
+$autoEncryptionOpts = [
+    'keyVaultClient' => $keyVaultClient,
+    'keyVaultNamespace' => 'default.keys',
+    'kmsProviders' => ['local' => ['key' => new MongoDB\BSON\Binary(str_repeat('0', 96), 0)]],
+];
+
+$manager = new MongoDB\Driver\Manager(URI, [], ['autoEncryption' => $autoEncryptionOpts, 'disableClientPersistence' => true]);
+
+$bulk = new MongoDB\Driver\BulkWrite();
+$bulk->insert(['x' => 1]);
+$bulk->insert(['x' => 2]);
+$bulk->insert(['x' => 3]);
+$manager->executeBulkWrite(NS, $bulk);
+
+MongoDB\Driver\Monitoring\addSubscriber(new CommandLogger);
+
+$query = new MongoDB\Driver\Query([], ['batchSize' => 2]);
+$cursor = $keyVaultClient->executeQuery(NS, $query);
+
+$childPid = pcntl_fork();
+
+if ($childPid === 0) {
+    /* Executing a query with the parent's client resets this client as well as
+     * the keyVaultClient. Continuing iteration of the cursor opened on the
+     * keyVaultClient before resetting it should then result in an error due to
+     * the client having been reset. */
+    $childCursor = $manager->executeQuery(NS, $query);
+
+    echo throws(
+        function () use ($cursor) { iterator_count($cursor); },
+        MongoDB\Driver\Exception\RuntimeException::class
+    ), "\n";
+
+    unset($childCursor);
+    echo "Child exits\n";
+    exit;
+}
+
+if ($childPid > 0) {
+    $waitPid = pcntl_waitpid($childPid, $status);
+
+    if ($waitPid === $childPid) {
+        echo "Parent waited for child to exit\n";
+    }
+
+    unset($cursor);
+}
+
+?>
+===DONE===
+<?php exit(0); ?>
+--EXPECT--
+Parent executes find with batchSize: 2
+Child executes listCollections
```

Comment:
Where does this come from, and why does it come between both `find` operations? Does libmongocrypt execute this when a Manager is initialized upon executing the first operation?